features/step_definitions/test_steps.rb

```diff
@@ -33,7 +33,7 @@
   add_named_record name, create(:test, options)
 end
 
-Given /^test "(.+)"(?: was last run by (.+) and)? has category (.+?)(?: and tags (.+?))?(?: and tickets (.+?))? for version (.+)$/ do |name,runner_name,category_name,tag_names,ticket_names,version_name|
+Given /^test "(.+)"(?: is (passing|failing)(?: and (active|inactive))?)?(?: was last run by (.+) and)? has category (.+?)(?: and tags (.+?))?(?: and tickets (.+?))? for version (.+)$/ do |name,passing,active,runner_name,category_name,tag_names,ticket_names,version_name|
```

Comment:
You should add an optional "and" before "was last run": `(?: and)?` so that the resulting sentence is more readable.

"And test "X" is passing and active was last run" is not good.

"And test "X" is passing and active and was last run" is better.